pkg/generate/app/cmd/template.go

```diff
@@ -38,9 +38,11 @@ func TransformTemplate(tpl *templateapi.Template, client client.TemplateConfigsN
 
 	// ensure the template objects are decoded
 	// TODO: in the future, this should be more automatic
-	if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
-		err = errors.NewAggregate(errs)
-		return nil, fmt.Errorf("error processing template %s: %v", name, err)
+	if decode {
+		if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
```

Comment:
but you can't run decodelist on https://github.com/openshift/origin/blob/master/test/testdata/template_required_params.yaml because that is a Template, not a List.